serverless/cmd/distribute/github/main.go

```diff
@@ -0,0 +1,305 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// feeder/github is a feeder implementation for GitHub Action based serverless distributors.
+//
+// This tool feeds checkpoints from a log into one or more witnesses, and then
+// raises a PR against the distributor repo to add any resulting additional
+// witness signatures.
+package main
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"gopkg.in/yaml.v2"
+
+	"github.com/google/trillian-examples/formats/log"
+	"github.com/google/trillian-examples/serverless/cmd/feeder/impl"
+	"github.com/google/trillian-examples/serverless/config"
+	"github.com/google/trillian-examples/serverless/internal/github"
+	"golang.org/x/mod/sumdb/note"
+
+	wit_http "github.com/google/trillian-examples/witness/golang/client/http"
+)
+
+const usage = `Usage:
+ distribute --distributor_repo --fork_repo --distributor_path --config_file --interval
+
+Where:
+ --distributor_repo is the repo owner/fragment from the distributor github repo URL.
+     e.g. github.com/AlCutter/serverless-test -> AlCutter/serverless-test
+ --fork_repo is the repo owner/fragment of the forked distributor to use for the PR branch.
+ --distributor_path is the path from the root of the repo where the distributor files can be found,
+ --config_file is the path to the config file for the serverless/cmd/feeder command.
+ --interval if set, the script will continuously feed and (if needed) create witness PRs sleeping
+     the specified number of seconds between attempts. If not provided, the tool does a one-shot feed.
+
+`
+
+var (
+	distributorRepo   = flag.String("distributor_repo", "", "The repo owner/fragment from the distributor repo URL.")
+	distributorBranch = flag.String("distributor_branch", "master", "The branch that PRs will be proposed against on the distributor_repo.")
+	forkRepo          = flag.String("fork_repo", "", "The repo owner/fragment from the feeder (forked distributor) repo URL.")
+	distributorPath   = flag.String("distributor_path", "", "Path from the root of the repo where the distributor files can be found.")
+	configPath        = flag.String("config_file", "", "Path to the config file for the serverless/cmd/feeder command.")
+	interval          = flag.Duration("interval", time.Duration(0), "Interval between checkpoints.")
```

Comment:
Maybe add something to the effect that the default setting of 0 will cause the program to distribute once and exit?

Reply:
Yeah, nice.